index.bs

```diff
@@ -519,21 +518,28 @@ operations in the graph onto a {{GPUCommandBuffer}}. The callers are responsible
+## Context and Device Association ## {#programming-model-context-device-association}
 
-In a situation when a GPU context executes a graph with a constant or an input in the system memory as an {{ArrayBufferView}}, the input content is automatically uploaded from the system memory to the GPU memory, and downloaded back to the system memory of an {{ArrayBufferView}} output buffer at the end of the graph execution. This data upload and download cycles will only occur whenever the execution device requires the data to be copied out of and back into the system memory, such as in the case of the GPU. It doesn't occur when the device is a CPU device. Additionally, the result of the graph execution is in a known layout format. While the execution may be optimized for a native memory access pattern in an intermediate result within the graph, the output of the last operation of the graph must convert the content back to a known layout format at the end of the graph in order to maintain the expected behavior from the caller's perspective.
+An {{MLContext}} interface represents a state of neural network execution. An important function of this state is manage resources used in the compilation 
+and execution of the neural network graph. An implementation in a user agent may implement this state in term of CPU resources when an {{MLContext}} is created 
+without an explicit association with a hardware device (aka. a *"default context"*). However, when an {{MLContext}} is created from a specific {{GPUDevice}} that 
+is also used by the application of WebGPU (aka. a *"GPU context"*), any GPU resource created from the same {{GPUDevice}} such as the {{GPUBuffer}} or {{GPUTexture}} 
+is considered a native resource that can also be used as a graph constant, input, or output operand.
 
-When an {{MLContext}} is created with {{MLContextOptions}}, the user agent selects and creates the underlying execution device by taking into account the application's [=power preference=] and [=device type=] specified in the {{MLPowerPreference}} and {{MLDeviceType}} options.
+In a situation when a GPU context executes a graph with a constant or an input or output allocated in the system memory as in an {{ArrayBufferView}}, the content 
+is automatically uploaded from the system memory to the GPU memory, and downloaded back to the system memory of an {{ArrayBufferView}} output buffer at the end of 
+the graph execution. This automatic data upload and download cycles will only occur whenever the executing GPU context determines that the data be copied out of or back 
```

Comment:
s/This automatic data/The automatic data

Also, if the following is a conformance requirement (in practice something for which we can write a test case for):

s/the data be/the data must be 

If it is informative prose instead, we usually try to avoid [RFC2119 terminology](https://www.w3.org/TR/webnn/#w3c-conventions) and could say:

s/the data be/the data has to be

If the latter, informative prose is expected to be wrapped into a "Note:" block. Alternative convention is to add "This section is non-normative." at the beginning of a section that in its entirety is non-normative (=informative), example: https://html.spec.whatwg.org/multipage/introduction.html#is-this-html5?

Reply:
This section of the new text doesn't describe new behavior, just repeat what was said earlier about automatic data upload/download to and from the GPU. Considering that the entire section contains the overall description of the programming model, I feel that this text naturally belongs here and not in a separate note section.

Reply:
Fixed